c10/core/DispatchKeySet.h

```diff
@@ -120,6 +120,65 @@ class DispatchKeySet final {
 private:
   DispatchKeySet(uint64_t repr) : repr_(repr) {}
   uint64_t repr_ = 0;
+
+public:
+  // STL iterator for DispatchKeySet. Iterates through all DispatchKeys in the
+  // set. The iterator is only invalidated by the destruction ofthe underlying
+  // DispatchKeySet as the iterator stores a pointer to the raw represenation of
+  // the DispatchKeySet.
+  class iterator {
+   public:
+    using self_type = iterator;
+    using iterator_category = std::input_iterator_tag;
```

Comment:
Input iterator is probably good enough for all forseeable use cases we have, but realistically you probably actually have a bidirectional iterator here!

Reply:
Agreed, with the current implementation it should be fairly easy to upgrade to a bidirectional iterator.